### PR TITLE
Cleanup debug log messages when loading stages

### PR DESCRIPTION
**Pull request by skshetry**
1. Simplified callback stage debug message
2. Reduced no lock entry message to trace
   If lock file does not have some entry, it will simply complain
   that lock file is not up to date once.
   If there is no lock file or empty lock file, it will complain that lock file not found.
3. Annoying parametrization related wdir-interpolated debug message removed.

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: dvc/stage/__init__.py [hunk 1]
```diff
@@ -256,12 +256,7 @@ def changed_deps(self):
             return False
 
         if self.is_callback:
```
Comment (efiop) [inline dvc/stage/__init__.py:258]: The whole `callback` logic can be deleted now, it has been obsoleted since 1.0

File: dvc/stage/__init__.py [hunk 2]
```diff
@@ -256,12 +256,7 @@ def changed_deps(self):
...
-            logger.debug(
-                '%s is a "callback" stage '
-                "(has a command and no dependencies) and thus always "
-                "considered as changed.",
-                self,
-            )
+            logger.debug("%s has no command and dependencies", self.addressing)
```
Comment (skshetry) [inline dvc/stage/__init__.py:259]:
```suggestion
            logger.debug("%s has a command but no dependencies", self.addressing)
```

File: dvc/stage/__init__.py (hunk 1 above)
Comment (skshetry) [reply dvc/stage/__init__.py:258]: How so? The callback is still present on the outputs during run/repro, so let's remove it later at the same time.

File: dvc/stage/__init__.py [hunk 3]
```diff
@@ -256,12 +256,7 @@ def changed_deps(self):
             return False
 
         if self.is_callback:
-            logger.debug(
-                '%s is a "callback" stage '
-                "(has a command and no dependencies) and thus always "
-                "considered as changed.",
-                self,
-            )
+            logger.debug("%s has a command but no dependencies", self.addressing)
```
Comment (skshetry) [inline dvc/stage/__init__.py:259]:
```suggestion
            logger.debug(
                "%s has a command but no dependencies",
                self.addressing
            )
```

File: dvc/stage/__init__.py (hunk 1 above)
Comment (efiop) [reply dvc/stage/__init__.py:258]: We have a checkbox https://github.com/iterative/dvc/issues/4841 . Sure, can do that separately, just saying.

**Review by efiop — APPROVED** [review]

**Pull request merged**